docs/core/deploying/trimming/prepare-libraries-for-trimming.md

```diff
@@ -8,360 +8,234 @@ ms.date: 06/12/2023
 
 # Prepare .NET libraries for trimming
```

Comment:
As a general comment - we've made some improvements in this are for .NET 8. And should update this doc to include .NET 8 - and recommend using it.

There's another consideration which is mentioned already but it might be worth pointing it out more prominently. Higher version SDK can build libraries targeting lower version TFMs and the higher version SDK will have fixes and improvements. So we should pretty much always recommend using the latest SDK available (ideally .NET 8 already). This has very direct impact on the trimming story - the examples here for the project files will look somewhat different (and hopefully simpler) for .NET 8.

Finally, recently we've learned that multi-targeting is not only very common, but basically a must have when enabling trimming. (This is based on experience trying to make some NuGets trim compatible). I think the document should have a section about the multi-targeting and our recommendations around it. But the core point is that very few libraries will only target latest TFM. By far the most common case is either library targets `netstandard2.0` or it multi-targets `net472`, `netstandard2.0` and `net6.0` (or some combination of these).

I'm writing all this down so that we don't forget - it may well need to be done in multiple PRs and updates to the docs. But to have a good consistent story for library trimming (and even more importantly for AOT compatibility), we will need to include .NET8 and the above topics.

Reply:
> Finally, recently we've learned that multi-targeting is not only very common, but basically a must have when enabling trimming. (This is based on experience trying to make some NuGets trim compatible). I think the document should have a section about the multi-targeting and our recommendations around it. But the core point is that very few libraries will only target latest TFM. By far the most common case is either library targets `netstandard2.0` or it multi-targets `net472`, `netstandard2.0` and `net6.0` (or some combination of these).

@vitek-karas  see #36730 
If you can get someone to draft it, I'll clean it up.